geode-core/src/main/java/org/apache/geode/metrics/MetricsPublishingService.java

```diff
@@ -77,16 +79,19 @@
 public interface MetricsPublishingService {
 
   /**
-   * Invoked when a metrics session is started during cache initialization. The implementation can
-   * use the metrics session to add sub-registries to the cache's composite registry for publishing
-   * metrics to external monitoring systems.
+   * Invoked when a metrics session starts. A typical implementation will create a meter registry
+   * that can publish to an external monitoring system, and call
+   * {@link MetricsSession#addSubregistry(MeterRegistry)} to add that registry to the session.
    *
-   * @param session the metrics session that is used to connect sub-registries
+   * @param session the metrics session whose metrics this publishing service will publish
    */
   void start(MetricsSession session);
 
   /**
-   * Invoked when a metrics session is stopped during cache close.
+   * Invoked when a metrics session stops. The implementation should call
+   * {@link MetricsSession#removeSubregistry(MeterRegistry)} to remove each meter registry it added
+   * to the session. The implementation should also clean up any other resources it allocated during
+   * {@link #start(MetricsSession)}.
    */
   void stop();
```

Comment:
This was an oversight on my part when I wrote the original code. I have regretted that ever since.

Perhaps now is the time to change it. Should we change this method, or deprecate it and add a new one?

Reply:
Go ahead and change it. It's still marked as Experimental.

Reply:
I've pushed a new commit that adds the parameter.